### DISCUSSION TITLE
Vercel Blob: No token found

### DISCUSSION THREAD

**Discussion by ThanushKrishna** [vercel/community#5159] [Feature Question]
### Question

Vercel Blob: No token found. Either configure the `BLOB_READ_WRITE_TOKEN` environment variable or pass a `token` option to your calls.

I am using Vercel blob in my nextjs app which is deployed in Vercel itself.
According to the documentation it's not needed to pass the token for any operations if the app is on Vercel and it works as expected when I am uploading the images. Still, when I am trying to delete a blob I get this error:

here is my handlefileDelete block:

 const handlefileDelete = async (index:number) => {
            
            try{

                if(links.length===1){
                    await del(links[0]); 
                    onSelectFile("");
                    setLinks([]);
                }
                if(links.length > 0){
                    console.log("Deleting Blob:" + links[index]);
                    await del(links[index], {
                        "token": process.env.BLOB_READ_WRITE_TOKEN
                    });                    
                    links.splice(index, 1);                         
                    onSelectFile(links.join(" "))      
                }
            }
            catch(e){
                console.log(e);
                return;
            }            
        }

**Comment by ThanushKrishna** [comment]
we are going live in 2 weeks with this feature, any support is appreciated.

**Comment by dtrejo112** [comment]
Hello I ran into this same problem when I tried performing a blob method on the client side, per the blob docs "To use the methods of the Blob SDK, you will need to call them inside an edge function, a serverless function or even a browser" . In my app directory I had to define a path like this api/func-name/route.ts and I placed the action I wanted to perform in this path. 

```
api/func-name/route.ts

import { list } from '@vercel/blob';
 
export const runtime = 'edge';
 
export async function GET() {
    const { folders, blobs } = await list({ mode: 'folded', prefix: 'project-images/' });
    return Response.json({ blobs });
}
```

Replace the GET() with whatever operation youd like to perform. 

Then on the client side: 

```
'use client';
import Image from "next/image";
import { list } from '@vercel/blob';
import { use, useEffect, useState } from "react";

const Projects = () => {
    const [isLoading, setIsLoading] = useState<boolean>(false);
    const [images, setImages] = useState<Blob[]>([]);

    useEffect(() => {
        getImages();
    }, []);

    const getImages = async () => {
        setIsLoading(true);
        try {
          const response = await fetch('api/get-images');
          const data = await response.json();
          console.log(data);
          setImages(data.blobs);
        } catch (error) {
          console.error(error);
        }
        setIsLoading(false);
      };
```

Hopefully this helps.
